src/main/java/com/ibm/cloud/sdk/core/security/IamToken.java

```diff
@@ -52,27 +54,46 @@ public Long getExpiration() {
   }
 
   /**
-   * Check if currently stored access token is valid.
+   * Returns true iff currently stored access token should be refreshed.
    *
-   * Using a buffer to prevent the edge case of the
-   * token expiring before the request could be made.
+   * This method uses a buffer to prevent the edge case of the
+   * token expiring before the request could be made, so this method will return true
+   * if the current time is within that buffer.
    *
    * The buffer will be a fraction of the total TTL. Using 80%.
    *
-   * @return true iff the current access token is valid and not expired
+   * This method also updates the expiration time if it determines the token needs refreshed to prevent other
+   * threads from making multiple refresh calls.
+   *
+   * @return true iff the current access token is invalid or past the expiration buffer
```

Comment:
I think you can use the exact language from `Cp4dToken` to describe the `@return`:
>    * @return true if token is invalid or past the refresh time, false otherwise